shared/src/main/scala/mouse/anyf.scala

```diff
@@ -49,4 +48,13 @@ final class AnyFOps[F[_], A](private val fa: F[A]) extends AnyVal {
   def liftOptionT(implicit F: Functor[F]): OptionT[F, A] =
     OptionT.liftF(fa)
 
-}
+  // applicativeError
+  def recoverAsLeft[E, L](pf: PartialFunction[E, L])(implicit F: ApplicativeError[F, E]): F[Either[L, A]] =
+    recoverEither(pf.andThen(Left(_)))
+
+  def recoverAsRight[E](pf: PartialFunction[E, A])(implicit F: ApplicativeError[F, E]): F[Either[Nothing, A]] =
+    recoverEither(pf.andThen(Right(_)))
+
+  def recoverEither[E, L](pf: PartialFunction[E, Either[L, A]])(implicit F: ApplicativeError[F, E]): F[Either[L, A]] =
+    F.recover(mapAsRight[L])(pf)
```

Comment:
When https://github.com/typelevel/mouse/pull/508 is published, It can be used as a counterpart:
```
val fa: F[Unit] = ???

fa.attempt.leftFlatMapOrKeepIn(pf)
```

But if we take allocations into consideration, it's probably better to have a distinct syntax for that.